### PR TITLE
Allow turning circuits into operators

### PR DESCRIPTION
**Pull request by jleni**
I still need to add a couple of unit tests, etc. and but I thought it made sense to submit an early PR to check if the approach meets your expectations.

You can check `examples/custom_operator.py` to see how the new feature is supposed to be used.

### CODE REVIEW

File: examples/custom_operator.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    """
+    S = Sgate(-1)
+    S | q[0]
+    S | q[1]
+    S | q[2]
+    S | q[3]
+
+
+@operator(ns=3)
+def circuit_op(v1, v2, q):
```
Comment (josh146) [inline examples/custom_operator.py]: Nice that it works with parameters! Does the operator decorator always require the qumode register `q` as the last argument?

Comment (jleni) [reply examples/custom_operator.py]: It works with parameters and also for class methods, like the example in the unit test: https://github.com/XanaduAI/strawberryfields/pull/36/files#diff-c487e7ec5c066aeaecbe95c769fdccaaR31

Yes, the last argument are always the register(s).

**Comment by josh146** [comment]
Hi Juan, thanks for your contribution! This is a really high quality PR - it's well linted, and the unittests are coming along well. Let me know when you have finished, and I'll do a short code review before merging.

**Comment by jleni** [comment]
Thanks! :) I would say the main functionality is there and it could be reviewed.
I am not sure if you would like to document the feature in this very same PR. But maybe it is a good idea that you review the code/approach before that.

**Comment by josh146** [comment]
Hmm, I'm not sure why Travis/Codecov didn't pick up this pull request. Let me see if I can trigger a build manually, and then I'll merge

**Comment by jleni** [comment]
Yes, I noticed that too. In my experience, circleci behaves a lot better than travis

**Comment by josh146** [comment]
Very odd. I have downloaded it locally, tested it, and checked the coverage, so I am happy to perform the merge now.

**Pull request merged**